location_server.py

```diff
@@ -55,11 +55,15 @@ def users_event():
 
 def clean_old_waypoints():
     now = datetime.datetime.now().timestamp()
-    for group_key in WAYPOINT_LOCS:
-        for zone in WAYPOINT_LOCS[group_key]:
+    for group_key in list(WAYPOINT_LOCS):
+        for zone in list(WAYPOINT_LOCS[group_key]):
             for waypoint in list(WAYPOINT_LOCS[group_key][zone]):
                 if now > waypoint[1]:
                     WAYPOINT_LOCS[group_key][zone].pop(waypoint)
+                    if len(WAYPOINT_LOCS[group_key][zone]) == 0:
```

Comment:
The pythonic way to do this is just check the truthiness since empty lists/ducts are False, rather than checking length.

Reply:
Pho sho, I just copied the logic from another function lower down. I'll update them both though real quick :)